gqlalchemy/loaders.py

```diff
@@ -0,0 +1,569 @@
+    """
+
+    label: Optional[str] = None
+    column_names_mapping: Dict[str, str] = field(default_factory=dict)
+
+    def get_property_name(self, column_name: str):
+        return self.column_names_mapping.get(column_name, column_name)
+
+
+class DataSource(ABC):
+    """
+    Base class. Inherit this class and encapsulate reading of the file format from some data source
+    """
+
+    def __init__(self, file_extension: str) -> None:
+        """
+        :param file_extension: Extension for which the reading is implemented
+        """
+        self._file_extension = file_extension
+
+    @abstractmethod
+    def load_data(self, collection_name: str, is_cross_table: bool = False) -> None:
+        """
+        Override this method in the derrived class. Intended to be used for reading data from data format.
+
+        :param collection_name: Name of the collection from which to read data
+        :param is_cross_table: Indicate whether or not the collection contains associative table (default=False)
+        """
+        pass
+
+
+class S3DataSource(DataSource):
+    """
+    Hold the implementation of reading files from Amazon S3
+    """
+
+    def __init__(
+        self,
+        bucket_name: str,
+        s3_access_key: str,
+        s3_secret_key: str,
+        s3_region: str,
+        file_extension: str,
+        s3_session_token: Optional[str] = None,
+    ) -> None:
+        """
+        :param bucket_name: Name of the bucket on S3 from which to read the data
+        :param s3_access_key: S3 access key
+        :param s3_secret_key: S3 secret key
+        :param s3_region: S3 region
+        :param file_extension: Extension for which the reading is implemented
+        :param s3_session_token: S3 session token (Optional)
+        """
+        super().__init__(file_extension=file_extension)
+        self._bucket_name = bucket_name
+        self._s3_config = {
+            "access_key": s3_access_key,
+            "secret_key": s3_secret_key,
+            "region": s3_region,
+            "session_token": s3_session_token,
+        }
+
+    def load_data(
+        self,
+        collection_name: str,
+        is_cross_table: bool = False,
+        columns: Optional[List[str]] = None,
+    ) -> None:
+        """
+        Read the data from S3 and process it in batches.
+
+        :param collection_name: Name of the collection from which to read the data
+        :param is_cross_table: Indicate whether or not the collection contains associative table (default=False)
+        :param columns: List of columns to be read from the collection (Optional)
+        """
+        s3 = fs.S3FileSystem(**self._s3_config)
+        source = f"{self._bucket_name}/{collection_name}.{self._file_extension}"
+        print("Loading data from " + ("cross " if is_cross_table else "") + f"table {source}...")
+
+        # Load dataset via Pyarrow
+        dataset = ds.dataset(source=source, filesystem=s3)
```

Comment:
need to specify `format=file_extension` parameter, default is parquet

Reply:
done